source/extensions/common/wasm/wasm.h

```diff
@@ -338,6 +368,14 @@ class Wasm : public Envoy::Server::Wasm,
   std::string code_;
   std::string initial_configuration_;
   bool allow_precompiled_ = false;
+
+  // Stats/Metrics
+  uint32_t next_counter_metric_id_ = kMetricTypeCounter;
```

Comment:
constexpr ?

Reply:
These are mutable.   They start with the say kMetricTypeCounter then are incremented by kMetricIdIncrement to get new counter specific indexes.

Reply:
Get it. Thanks!